src/extension.ts

```diff
@@ -388,7 +388,7 @@ class ConventionalCommitMessage {
   private static shouldSkip(
     czConfig: CzConfig | undefined,
     messageType: string
-  ): czConfig is CzConfig {
+  ): boolean {
```

Comment:
Why did you change this? The typescript 'x is y' return implies a boolean result as well as narrowing the type.
Effectivity you are weakening the compiler here.

Reply:
Understood. sorry for that

Reply:
Nothing to be sorry for. 😃

Reply:
changing that is causing some linting issues:
```
 error TS2339: Property 'scopes' does not exist on type 'never'.

480         if (this.czConfig.scopes && this.czConfig.scopes[0] !== undefined) {
```

Reply:
> changing that is causing some linting issues:
> 
> ```
>  error TS2339: Property 'scopes' does not exist on type 'never'.
> 
> 480         if (this.czConfig.scopes && this.czConfig.scopes[0] !== undefined) {
> ```


this doesn't happen when the return type is `boolean`

Reply:
The error means the type is not possible and would result in never executed code.
I guess the typing is incorrect. I take a look later on.

Reply:
okay thanks for your time! I am a typescript noob :)

Reply:
After having a look I have to say sorry :smiley: 
Your first change wasn't that bad. The code is quiet old and TypeScript evolved. I thing it might need a complete overhaul to be more understandable.
Please change it it `boolean` again. I'm happy to merge it.

Reply:
sure :) but I think the return type is `boolean` at the current state

Reply:
Hello @KnisterPeter, any news?